pypeit/core/wavecal/identify.py

```diff
@@ -0,0 +1,770 @@
+        disp_est = self.fitsol_deriv()
+        for wav in range(wave_est.size):
+            if self._lineflg[wav] == 1:
+                # User has manually identified this line already
+                continue
+            pixdiff = np.abs(wave_est[wav]-self._lines)
+            amin = np.argmin(pixdiff)
+            pxtst = pixdiff[amin]/disp_est[wav]
+            self._lineids[wav] = self._lines[amin]
+            if pxtst < self.par['match_toler']:
+                # Acceptable
+                self._lineflg[wav] = 1
+            else:
+                # Unacceptable
+                self._lineflg[wav] = 3
+        # Now that we've automatically identified lines, update the canvas
+        self.replot()
+
+    def delete_line_id(self):
+        """ Remove an incorrect line ID
+        """
+        rmid = self.get_detns()
+        self._lineids[rmid] = 0.0
+        self._lineflg[rmid] = 0
+
+    def fitsol_value(self, xfit=None, idx=None):
+        """ Calculate the wavelength at a pixel
+
+        Parameters
+        ----------
+        xfit : ndarray, float
+          pixel values that the user wishes to evaluate the wavelength
+        idx : int
+          Index of the arc line detections that the user wishes to evaluate the wavelength
+
+        Returns
+        -------
+        disp : ndarray, float, None
+          The wavelength (Angstroms) of the requested pixels
+        """
+        if xfit is None:
+            xfit = self._detns
+        if self._fitdict['coeff'] is not None:
+            if idx is None:
+                return np.polyval(self._fitdict["coeff"], xfit / self._fitdict["scale"])
+            else:
+                return np.polyval(self._fitdict["coeff"], xfit[idx] / self._fitdict["scale"])
+        else:
+            msgs.bug("Cannot predict wavelength value - no fit has been performed")
+            return None
+
+    def fitsol_deriv(self, xfit=None, idx=None):
+        """ Calculate the dispersion as a function of wavelength
+
+        Parameters
+        ----------
+        xfit : ndarray, float
+          pixel values that the user wishes to evaluate the dispersion
+        idx : int
+          Index of the arc line detections that the user wishes to evaluate the dispersion
+
+        Returns
+        -------
+        disp : ndarray, float, None
+          The dispersion (Angstroms/pixel) as a function of wavelength
+        """
+        if xfit is None:
+            xfit = self._detns
+        if self._fitdict['coeff'] is not None:
+            cder = np.polyder(self._fitdict["coeff"])
+            if idx is None:
+                return np.polyval(cder, xfit / self._fitdict["scale"]) / self._fitdict["scale"]
+            else:
+                return np.polyval(cder, xfit[idx] / self._fitdict["scale"]) / self._fitdict["scale"]
+        else:
+            msgs.bug("Cannot predict wavelength value - no fit has been performed")
+            return None
+
+    def fitsol_fit(self):
```

Comment:
is this method used?  or are you just doing calls to 
`fitting.iterative_fitting()` elsewhere?

Reply:
Is there a reason not to use iterative fitting? That algorithm also allows you to choose the order. I'm slightly concerned that the line-IDs are based on a different fitting approach, since iterative fitting is so robust.